### PR TITLE
Dmytro/int 1756 merged  app and  document content should be moved to layout

### PR DESCRIPTION
**Pull request by dmytro068**

### CODE REVIEW

File: next/13/app-directory-boilerplate/index.ts [hunk 1]
```diff
@@ -52,9 +50,10 @@ export default function NotFound() {
 }
 `;
 
-enum FilePurpose {
+const enum FilePurpose {
 	// root directory
 	ROOT_LAYOUT = 'ROOT_LAYOUT',
+	ROOT_LAYOUT_COMPONENT = 'ROOT_LAYOUT_COMPONENT',
```
Comment (grzpab) [inline next/13/app-directory-boilerplate/index.ts:56]: what's that for?

File: next/13/app-directory-boilerplate/index.ts [hunk 2]
```diff
@@ -599,62 +599,76 @@ const updateLayoutComponent = (sourceFile: SourceFile) => {
 	}`);
 };
 
-const extractStatements = (sourceFile: SourceFile) => {
-	const statements = sourceFile.getStatements();
+const findComponent = (sourceFile: SourceFile) => {
```
Comment (grzpab) [inline next/13/app-directory-boilerplate/index.ts:602]: what's the returned type from it?

File: next/13/app-directory-boilerplate/index.ts [hunk 3]
```diff
@@ -599,62 +599,76 @@ const updateLayoutComponent = (sourceFile: SourceFile) => {
...
+	sourceFile: SourceFile,
+) => {
+	const component = findComponent(sourceFile);
+
+	if (component === null) {
+		return;
+	}
+
+	if (!Node.isArrowFunction(component)) {
+		component.rename('LayoutClientComponent');
```
Comment (grzpab) [inline next/13/app-directory-boilerplate/index.ts:658]: can you explain this line?

File: next/13/app-directory-boilerplate/test.ts [hunk 4]
```diff
@@ -114,7 +114,7 @@ const getServerSideProps = () => {
 };
 `;
 
-describe('next 13 app-directory-boilerplate', function () {
+describe.only('next 13 app-directory-boilerplate', function () {
```
Comment (grzpab) [inline next/13/app-directory-boilerplate/test.ts:117]: can you remove `.only`?

File: next/13/app-directory-boilerplate/index.ts (hunk 3 above)
Comment (DmytroHryshyn) [reply next/13/app-directory-boilerplate/index.ts:658]: im taking _app file and based on it i build layout client component, so i want to change original _app component name to more appropriate

File: next/13/app-directory-boilerplate/index.ts (hunk 1 above)
Comment (grzpab) [reply next/13/app-directory-boilerplate/index.ts:56]: ah I see, makes sense.

**Pull request merged**